src/cmd/create.go

```diff
@@ -718,20 +743,89 @@ func pullImage(image, release string) (bool, error) {
 		return false, nil
 	}
 
+pull_image:
```

Comment:
We should handle the authentication errors in the same way we handle fallbacks in `run`. ie., refactor the guts of `pullImage` into smaller functions and then run a loop.

Using `goto` to loop across such a big chunk of code with some `return`s in between is soon going to turn into a nightmare.

Reply:
You are absolutely right. I was not sure how to properly separate this, so I went with the `goto` option. I'll try to think of something.